### ISSUE TITLE
Roundcube unable to send outbound email smtp 553 error [postfix/dovecot]

### ISSUE THREAD

**Issue by dundir** [roundcube/roundcubemail#5907]
Completely fresh install for postfix/dovecot/roundcube.  A similar setup can be found from the arstechnica tutorials for reproducibility purposes.

Postfix/Dovecot tested and verified working (can receive and send email via mail. Setup permits trusted networks (localhost) to send smtp email without authentication. Encryption (SSL) is enabled on external facing ports and services.

Roundcube consistently fails at sending outbound email. The error is a 553 user not logged in error. Users login through the web interface (Dovecot -IMAP) can receive, read, but not send emails. 

I've spent about a week trying to work around this problem and must conclude its a bug within the suite somewhere. There does not appear to be a workaround without hardcoding values into non-config files (breaking properly working authentication) or turning the mail server into an open relay. Neither option is suitable. 

The error occurs regardless of whether the variables; smtp_{server, port, user, pass} are defined, undefined, or left blank in any variation thereof; it doesn't seem to matter.

I've attached the logs with debug enabled as well as the configuration being used at the bottom. 

--FEEDBACK
I'll be moving on to another webmail suite for now, but I'm letting you know about the problem so it can be fixed.

Additionally, and this is more feedback than anything else, you may want to ...
1. have an example configuration file posted somewhere (its only ever referenced to the "auto-generated config"),
2. add a complete listing of configuration parameters and the parameter dependencies to the documentation
3. add more information to the auto-generated stub config

As it stands, the documentation lacks parameter dependency information; how to enable smtp_debug (a parameter not even listed in the configuration/installation section); the auto-generated config file is mostly blank; and this all combines to make successful troubleshooting require an almost intimate knowledge of your project. 

I'm sure you have better things to do than spend your time on dealing with troubleshooting issues that appear to revolve around syntax within a config file; I certainly do. 

Better to offload to the end user and provide them the tools so they can troubleshoot it effectively themselves but then that's just my two cents. Best Regards.

Sanitized Logs:
[Roundcube Logs.txt](https://github.com/roundcube/roundcubemail/files/1232315/Roundcube.Logs.txt)

Configuration:
[rc_mailconfig.txt](https://github.com/roundcube/roundcubemail/files/1232508/rc_mailconfig.txt)

**Comment by alecpl** [comment]
All you need is smtp_host, smtp_port and smtp_user='%u' and smtp_pass='%p'. This is what we have in defaults.inc.php:
```
// SMTP server host (for sending mails).
// Enter hostname with prefix tls:// to use STARTTLS, or use
// prefix ssl:// to use the deprecated SSL over SMTP (aka SMTPS)
// Supported replacement variables:
// %h - user's IMAP hostname
// %n - hostname ($_SERVER['SERVER_NAME'])
// %t - hostname without the first part
// %d - domain (http hostname $_SERVER['HTTP_HOST'] without the first part)
// %z - IMAP domain (IMAP hostname without the first part)
// For example %n = mail.domain.tld, %t = domain.tld
$config['smtp_server'] = 'localhost';

// SMTP port (default is 25; use 587 for STARTTLS or 465 for the
// deprecated SSL over SMTP (aka SMTPS))
$config['smtp_port'] = 25;

// SMTP username (if required) if you use %u as the username Roundcube
// will use the current username for login
$config['smtp_user'] = '';

// SMTP password (if required) if you use %p as the password Roundcube
// will use the current user's password for login
$config['smtp_pass'] = '';
```
I'd leave this ticket for considerations on a documentation or defaults improvements.

**Comment by dundir** [comment]
As a follow-up; and apologies if my initial feedback was a bit terse. I did manage to get a working instance of RC up and running.

There were two issues ultimately that broke my installation though there were more than a few false leads in getting down to the exact cause. 

The first revolved around php7 and the common-name of the in-use certificate not matching with the local loopback address being used for tls smtp/imap connections. The following settings corrected that issue and hopefully it can be helpfully included in the auto-generated config as a comment moving forward. This appears to be a problem for php versions 5.6+.

$config['imap_conn_options'] = array(
'ssl'=>array(
'verify_peer'=>false,
'verify_peer_name'=>false,
),);

$config['smtp_conn_options'] = array(
'ssl'=>array(
'verify_peer'=>false,
'verify_peer_name'=>false,
),);

I was unable to get RC sending email without TLS while I was tackling the issue though I suspect it was caused by the server not advertising AUTH PLAIN LOGIN without STARTTLS.

The second issue had to do with strange postfix behavior; which was ironically the result of a comment at the end of a line. ( smtpd_tls_mandatory_ciphers = !SSLv2 !SSLv3 \#POODLE )

It seemed like there was an issue with RC generating a correct AUTH base64 string before I stumbled across the above (by recreating main.cf from scratch) and subsequently the issue resolved itself.

I hope follow-up was helpful.  Best Regards
